src/main.rs

```diff
@@ -303,10 +310,15 @@ fn main() -> anyhow::Result<()> {
         let replacements = find_renames(&input_files, &new_files)?;
         println!();
 
-        check_for_existing_files(&replacements)?;
-        print_replacements(&replacements, opts.pretty_diff);
+        let check_existing = check_for_existing_files(&replacements);
+
+        match check_existing {
+            Ok(()) => print_replacements(&replacements, opts.pretty_diff),
+            e @ Err(_) if opts.assume_yes => return e,
+            Err(_) => {}
+        };
 
-        match prompt(opts.assume_yes)? {
+        match prompt(opts.assume_yes, check_existing.is_ok())? {
```

Comment:
I think the prompt shouldn't have that much logic, also the `is_successful` parameters name is confusing in my opinion.
What about this alternative: move the `possible_selection` vec outside of the `prompt` function, and simply give it as a parameter?
This way:
1. The possibles selections stays at the same place where we check the results, I think it's better for locality.
2. The prompt function stays simple, no specific logic based on obscure unknown condition (from the point of view of the function). 
3. Since you know beforehand the various conditions you can request prompt only with the relevant selection!
4. The prompt function uses the first possibility as the default, maybe for the case of confocts, the default should be Edit, not No? 

Other things I'm thinking:
- for the `possible_selection` vec, does it have to be a vec? We never try to modify it, so it could be a basic array, no?
- for the choices, I think it would be more `rusty`/`rust-like` to use an enum here, not strings. (but it might not be ideal though, if the thing that does the prompt require strings..)